components/apimgt/org.wso2.carbon.apimgt.micro.gateway/org.wso2.carbon.apimgt.micro.gateway.common/src/main/java/org/wso2/carbon/apimgt/micro/gateway/common/api/APIExecutionHandler.java

```diff
@@ -0,0 +1,104 @@
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.carbon.apimgt.micro.gateway.common.api;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.model.API;
+import org.wso2.carbon.apimgt.api.model.APIIdentifier;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.executors.APIExecutor;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+import org.wso2.carbon.apimgt.micro.gateway.common.dao.OnPremiseGatewayDAO;
+import org.wso2.carbon.apimgt.micro.gateway.common.exception.OnPremiseGatewayException;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.governance.api.generic.GenericArtifactManager;
+import org.wso2.carbon.governance.api.generic.dataobjects.GenericArtifact;
+
+import org.wso2.carbon.registry.core.Resource;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.registry.core.jdbc.handlers.RequestContext;
+
+import java.util.Map;
+
+/**
+ * This class consists methods that will create, prototype, publish, block, deprecate and
+ * retire  an API to API Manager.
+ * <p/>
+ * This executor used to publish a service to API store as a API.
+ *
+ */
+public class APIExecutionHandler extends APIExecutor {
+
+    Log log = LogFactory.getLog(APIExecutionHandler.class);
+
+    /**
+     * This method is called when the execution class is initialized.
+     * All the execution classes are initialized only once.
+     *
+     * @param parameterMap Static parameter map given by the user.
+     *                     These are the parameters that have been given in the
+     *                     lifecycle configuration as the parameters of the executor.
+     */
+    public void init(Map parameterMap) {
+    }
+
+    /**
+     * @param context      The request context that was generated from the registry core.
+     *                     The request context contains the resource, resource path and other
+     *                     variables generated during the initial call.
+     * @param currentState The current lifecycle state.
+     * @param targetState  The target lifecycle state.
+     * @return Returns whether the execution was successful or not.
+     */
+    public boolean execute(RequestContext context, String currentState, String targetState) {
+        APIExecutor apiExecutor = new APIExecutor();
```

Comment:
This is the super class, Can't we use super.execute() in line 101 without re initializing